lib/events.php

```diff
@@ -225,3 +225,40 @@ function widget_manager_update_group_event_handler($event, $object_type, $object
 		}
 	}
 }
+
+/**
+ * Remove fixed widgets from dashboard when parent widget is deleted from
+ * default widgets editor
+ *
+ * @param string     $event  "delete"
+ * @param string     $type   "object"
+ * @param ElggWidget $entity Widget
+ * @return void
+ */
+function widget_manager_remove_fixed_widgets($event, $type, $entity) {
```

Comment:
I guess it really depends on the use case. Theoretically, users can just add them if they want to again. In my project, users can't edit the layout and all widgets are kept in sync. Up to you really.